File: 9-PalindromeNumber/9-PalindromeNumber.rb
# Last updated: 4/8/2026, 2:09:16 AM
# @param {Integer} x
# @return {Boolean}
def is_palindrome(number)
    string = number.to_s
     reverse_str = ''
      x = string.length - 1
      while x >= 0
        reverse_str += string[x]
        x -= 1
      end
      if reverse_str == string 
        return true
      else
        return false
      end
end

File: 9-PalindromeNumber/9-PalindromeNumber.cpp
// Last updated: 4/8/2026, 2:09:18 AM
class Solution {
public:
    bool isPalindrome(int x) {
          bool isPal = true;

          if(x < 0){
            isPal = false;
            return isPal;
          }
          else if(x == 0){
            return isPal;
          }

          std::string str;
          std::stringstream ss;
          ss << x;
          ss >> str;

          for(int i = 0; i < str.length()/2; i++){
            if(str[i] != str[str.length()-1-i]){
              isPal = false;
              break;
            }
          }

          return isPal;
    }
};